library/core/tests/fmt/float.rs

```diff
@@ -5,7 +5,7 @@ fn test_format_f64() {
     assert_eq!("10", format!("{:.0}", 9.9f64));
     assert_eq!("9.8", format!("{:.1}", 9.849f64));
     assert_eq!("9.9", format!("{:.1}", 9.851f64));
-    assert_eq!("1", format!("{:.0}", 0.5f64));
+    assert_eq!("0", format!("{:.0}", 0.5f64));
```

Comment:
Suggestion: in addition to this change here, add s new `test_format_rounds_ties_to_even` test with a whole bunch of examples like those now in the PR discussion.  Just as a pseudo-documentation way of communicating to anyone tweaking things in future that, yes, this definitely is supposed to be working like this.

Reply:
Thanks, have added these for `f32` and `f64`, including some exponentials for good measure. Let me know if you want any specific additional examples.